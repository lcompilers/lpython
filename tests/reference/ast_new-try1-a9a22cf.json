{
    "basename": "ast_new-try1-a9a22cf",
    "cmd": "lpython --show-ast --new-parser --no-color {infile} -o {outfile}",
    "infile": "tests/parser/try1.py",
    "infile_hash": "a7fac6105109bebf6a38bd317f4bbf5876e2dd62a341c71232791619",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "ast_new-try1-a9a22cf.stdout",
    "stdout_hash": "2d208d18296ba0b82463ef7c49b2dedaed5d5600b0956f27084b454e",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}