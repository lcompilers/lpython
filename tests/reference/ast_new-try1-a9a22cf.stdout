(Module
    [(For
        (Name
            i
            Store
        )
        (Call
            (Name
                range
                Load
            )
            [(ConstantInt
                3
                ()
            )
            (ConstantInt
                0
                ()
            )
            (UnaryOp
                USub
                (ConstantInt
                    1
                    ()
                )
            )]
            []
        )
        [(Try
            [(Expr
                (Call
                    (Name
                        print
                        Load
                    )
                    []
                    []
                )
            )]
            [(ExceptHandler
                (Name
                    IOError
                    Load
                )
                ()
                [(If
                    (Compare
                        (Name
                            i
                            Load
                        )
                        Eq
                        [(ConstantInt
                            1
                            ()
                        )]
                    )
                    [(Raise
                        ()
                        ()
                    )]
                    []
                )
                (Expr
                    (Call
                        (Name
                            print
                            Load
                        )
                        [(ConstantStr
                            "retry"
                            ()
                        )]
                        []
                    )
                )]
            )]
            [(Break)]
            []
        )]
        []
        ()
    )
    (Try
        [(Pass)]
        [(ExceptHandler
            (Name
                ValueError
                Load
            )
            ()
            [(Pass)]
        )
        (ExceptHandler
            (Name
                TypeError
                Load
            )
            ()
            [(Pass)]
        )
        (ExceptHandler
            (Name
                ZeroDivisionError
                Load
            )
            ()
            [(Pass)]
        )
        (ExceptHandler
            ()
            ()
            [(Pass)]
        )]
        []
        []
    )
    (Try
        [(Assign
            [(Name
                a
                Store
            )]
            (ConstantInt
                1
                ()
            )
            ()
        )
        (If
            (Compare
                (Name
                    a
                    Load
                )
                LtE
                [(ConstantInt
                    0
                    ()
                )]
            )
            [(Raise
                (Call
                    (Name
                        ValueError
                        Load
                    )
                    [(ConstantStr
                        "That is not a positive number!"
                        ()
                    )]
                    []
                )
                ()
            )]
            []
        )]
        [(ExceptHandler
            (Name
                ValueError
                Load
            )
            ve
            [(Expr
                (Call
                    (Name
                        print
                        Load
                    )
                    [(Name
                        ve
                        Load
                    )]
                    []
                )
            )]
        )]
        []
        []
    )
    (Try
        [(Raise
            (Name
                KeyboardInterrupt
                Load
            )
            ()
        )]
        []
        []
        [(Expr
            (Call
                (Name
                    print
                    Load
                )
                [(ConstantStr
                    "Goodbye, world!"
                    ()
                )]
                []
            )
        )]
    )
    (Try
        [(Expr
            (ConstantEllipsis
                ()
            )
        )]
        [(ExceptHandler
            (Name
                Exception
                Load
            )
            e
            [(Expr
                (ConstantEllipsis
                    ()
                )
            )]
        )]
        []
        []
    )
    (Try
        [(Expr
            (ConstantEllipsis
                ()
            )
        )]
        [(ExceptHandler
            ()
            ()
            [(Expr
                (ConstantEllipsis
                    ()
                )
            )]
        )]
        [(Expr
            (ConstantEllipsis
                ()
            )
        )]
        [(Expr
            (ConstantEllipsis
                ()
            )
        )]
    )
    (Try
        [(Expr
            (Call
                (Attribute
                    (Name
                        shutil
                        Load
                    )
                    rmtree
                    Load
                )
                [(Call
                    (Attribute
                        (Attribute
                            (Name
                                os
                                Load
                            )
                            path
                            Load
                        )
                        join
                        Load
                    )
                    [(Name
                        asv_build_cache
                        Load
                    )
                    (Name
                        c
                        Load
                    )]
                    []
                )]
                []
            )
        )]
        [(ExceptHandler
            (Name
                OSError
                Load
            )
            ()
            [(Pass)]
        )]
        []
        []
    )]
    []
)
